packages/material-ui/src/Badge/Badge.js

```diff
@@ -23,21 +25,18 @@ export const styles = theme => ({
     alignContent: 'center',
     alignItems: 'center',
     position: 'absolute',
-    top: 0,
-    right: 0,
     boxSizing: 'border-box',
     fontFamily: theme.typography.fontFamily,
     fontWeight: theme.typography.fontWeightMedium,
     fontSize: theme.typography.pxToRem(12),
-    minWidth: RADIUS * 2,
+    minWidth: RADIUS_STANDARD * 2,
+    lineHeight: 1,
     padding: '0 4px',
```

Comment:
Could we compromise with:

```suggestion
    padding: '0 3px',
```
?

It allows the 2 digit badge to be round, without noticeably impacting the 3 pill-shaped ones...

Reply:
:+1:

Reply:
I have missed the end of the sentence in your feedback. I was so obssessed with the wrong vertical alignment that I didn't notice the round aspect.

Reply:
I had a closer look at the issue, I think that we should stick with 4 pixels because it helps better read the content (more background contrast).

Reply:
I'm not sure I follow. How does the padding affect the contrast?

If that isn't the best fix, what changed to break it in the first place, and let's fix that.

Reply:
From the limited numbers of tests that I have run, the badge background surface impacts the contrast. The problem is that both the body and the badge text are white. If the badge doesn't have enough surface, the text won't contrast enough.

I'm wondering, is the fact that the badge is not round a problem on its own? I believe that the shape has changed when we introduced the support for long badges.

Reply:
> If the badge doesn't have enough surface, the text won't contrast enough.

From my (also limited) testing, 3px seems perfectly readable.

> I'm wondering, is the fact that the badge is not round a problem on its own?

IMHO it should be round for up to two characters, otherwise it looks "off". For the longer text it's fine, because 3+ characters are clearly pill-shaped, with two characters it's "not quite round".

> I believe that the shape has changed when we introduced the support for long badges.

Ah that was it. Padding so that longer text doesn't bleed to the edge of the badge. Well then we for two character badges we are no worse off than before that change was made if we reduce it to `3px`. Making it conditional on the string length seems overkill.

Reply:
Looking at the diff, the higher the value, the more readable. What do you think of 5px? So we clearly see that it's not round?

**8px**
![Capture d’écran 2019-08-31 à 12 58 07](https://user-images.githubusercontent.com/3165635/64063051-334fc980-cbef-11e9-990f-bb219cb1039d.png)

**7px**
![Capture d’écran 2019-08-31 à 13 00 32](https://user-images.githubusercontent.com/3165635/64063059-5c705a00-cbef-11e9-8225-135a8c0f73bc.png)

**5px**
![Capture d’écran 2019-08-31 à 12 59 17](https://user-images.githubusercontent.com/3165635/64063052-33e86000-cbef-11e9-9eda-f4a39cf00061.png)

**4px**
![Capture d’écran 2019-08-31 à 12 55 03](https://user-images.githubusercontent.com/3165635/64063009-bc1a3580-cbee-11e9-9343-18e2de45615e.png)

**3px**
![Capture d’écran 2019-08-31 à 12 55 21](https://user-images.githubusercontent.com/3165635/64063010-bc1a3580-cbee-11e9-8aa8-1b822efc8c20.png)

> Alternatively we could make the text smaller.

I have explored it a bit, a font size of 12px seems the smaller value we should allow across the whole library (font-size: 12px, padding 3px looks more readable than font-size: 11px, padding 4px).

Reply:
Ok. If we're going for "clearly not round" then 6 or higher seems to be better. Since 6px is the median of the ones you benchmarked, and the smallest deviation from the current, let's go with that.

(Aside: Ant's scroll animation when the badge content changes is neat!)

Reply:
Ok, 6px sounds great :)